Mathlib/CategoryTheory/Adjunction/Mates.lean

```diff
@@ -108,13 +111,58 @@ def transferNatTrans : (G ⋙ L₂ ⟶ L₁ ⋙ H) ≃ (R₁ ⋙ G ⟶ H ⋙ R
       h.naturality, -Functor.map_comp, ← Functor.map_comp_assoc G, R₂.map_comp]
 #align category_theory.transfer_nat_trans CategoryTheory.transferNatTrans
 
+-- ER: A new construction with a new name.
+def Mates :
+    (G ⋙ L₂ ⟶ L₁ ⋙ H) ≃ (R₁ ⋙ G ⟶ H ⋙ R₂) where
+      toFun := by
+        intro α
+        have R₁Gη₂ := whiskerLeft (R₁ ⋙ G) adj₂.unit
+        have R₁αR₂ := whiskerRight (whiskerLeft R₁ α) R₂
+        have ε₁HR₂ := whiskerRight adj₁.counit (H ⋙ R₂)
+        exact R₁Gη₂ ≫ R₁αR₂ ≫ ε₁HR₂
+      invFun := by
+        intro β
+        have η₁GL₂ := whiskerRight adj₁.unit (G ⋙ L₂)
+        have L₁βL₂ := whiskerRight (whiskerLeft L₁ β) L₂
+        have HR₂ε₂ := whiskerLeft (L₁ ⋙ H) adj₂.counit
+        exact η₁GL₂ ≫ L₁βL₂ ≫ HR₂ε₂
+      left_inv := by
+        intro α
+        ext
+        unfold whiskerRight whiskerLeft
+        simp only [comp_obj, id_obj, Functor.comp_map, comp_app, map_comp, assoc, counit_naturality,
+          counit_naturality_assoc, left_triangle_components_assoc]
+        rw [← assoc, ← Functor.comp_map, α.naturality, Functor.comp_map, assoc, ← H.map_comp, left_triangle_components, map_id]
+        simp only [comp_obj, comp_id]
+      right_inv := by
+        intro β
+        ext
+        unfold whiskerLeft whiskerRight
+        simp only [comp_obj, id_obj, Functor.comp_map, comp_app, map_comp, assoc,
+          unit_naturality_assoc, right_triangle_components_assoc]
+        rw [← assoc, ← Functor.comp_map, assoc, ← β.naturality, ← assoc, Functor.comp_map, ← G.map_comp, right_triangle_components, map_id, id_comp]
+
+-- ER: Note these definitions agree.
+theorem RightMateEqualsTransferNatTrans
+    (α : G ⋙ L₂ ⟶ L₁ ⋙ H) :
+    Mates adj₁ adj₂ α = (transferNatTrans adj₁ adj₂) α := by
+  ext; unfold Mates transferNatTrans; simp
+
+-- ER: Note these definitions agree.
+theorem LeftMateEqualsTransferNatTrans.symm
+    (α : R₁ ⋙ G ⟶ H ⋙ R₂) :
+    (Mates adj₁ adj₂).symm α = (transferNatTrans adj₁ adj₂).symm α := by
+  ext; unfold Mates transferNatTrans; simp
+
+-- ER: It's not clear to me what this theorem is for.
```

Comment:
This theorem and the below (particularly in the Self case) are useful for proving things about the mate in applications - the self version for both are used to simplify proofs about cartesian closed-ness

Reply:
Thanks for the explanation. Would you mind if I renamed these as well (and propagated the change forward to other files?

Reply:
@b-mehta I've done this now. Let me know if you want me to change anything (eg the names).